homeassistant/components/spotify/recommendation_handling.py

```diff
@@ -61,7 +61,7 @@ def _get_current_weather_search_string(self, hass: HomeAssistant) -> str | None:
 
         current_weather_search_string = None
 
-        weather_entity_ids = self.get_entity_ids(hass, "weather")
+        weather_entity_ids = HolidayDateMapper().get_entity_ids(hass, "weather")
```

Comment:
Does not make sense that the weather feature uses a method i the HolidayDateMapper, I can try to fix it

Reply:
It is fixed in this pr now